packages/orchestrator-ui-components/src/components/WfoTable/WfoTableWithFilter/updateQueryString.ts

```diff
@@ -0,0 +1,52 @@
+const getValueWithoutBrackets = (value: string) => {
+    if (value.startsWith('(') && value.endsWith(')')) {
+        return value.slice(1, -1);
+    }
+    return value;
+};
+
+const isContainingSpaces = (value: string) => {
+    return value.indexOf(' ') !== -1;
+};
+
+const toQueryValue = (value: string) => {
+    return isContainingSpaces(value) ? `"${value}"` : value;
+};
+
+// Three cases to consider when this function is called
+// 1 - queryString is empty
+// 2 - fieldName already exists in query string
+// 3 - fieldName does not exist in query string
+export const updateQueryString = (
+    queryString: string | undefined,
+    fieldName: string,
+    value: string,
+) => {
+    // 1 - Empty query string returning directly fieldName:value
+    if (!queryString) {
+        return `${fieldName}:${toQueryValue(value)}`;
+    }
+
+    const fieldRegex = new RegExp(
+        `${fieldName}:(\\([\\S "]+\\)|"[\\S ]+"|\\S+)`,
```

Comment:
This also includes spaces in the "value" part. It's probably simpler to do the following:

Search for `"${fieldName}:"`
If no match => `//  1`

Otherwise, check if the value is a ValueGroup (starts with `(`)
If it is, find the first closing paren `)` and insert `| toQueryValue(value)` in front of it
If it isn't a ValueGroup, make it a valueGroup by enclosing the existing value in parentheses and insert `| toQueryValue(value)`

Reply:
Updates regex to match backend